src/CustomElements/v1/CustomElements.js

```diff
@@ -237,11 +243,21 @@ var CustomElementDefinition;
         new (definition.constructor)();
         console.assert(this._newInstance == null);
       }
-      if (definition.attributeChangedCallback && definition.observedAttributes.length > 0) {
+
+      var observedAttributes = definition.observedAttributes;
```

Comment:
Changed this to lookup the property only once since there's multiple lookups below.